pyxem/utils/indexation_utils.py

```diff
@@ -923,3 +923,527 @@ def peaks_from_best_vector_match(single_match_result, library, rank=0):
+        corresponding correlation values
+    """
+    polar_image = image_to_polar(image, delta_r, delta_theta, find_maximum=find_direct_beam, **kwargs)
+    polar_image = polar_image / np.sqrt(np.sum(polar_image**2))
+    positions, intensities = _simulations_to_arrays(simulations)
+    r_templates, theta_templates = _cartesian_positions_to_polar(positions[:,0], positions[:,1],
+                                                                 delta_r, delta_theta)
+    if keep >= 1.:
+        fraction = max((theta_templates.shape[0] - keep) / theta_templates.shape[0], 0.)
+    elif 0. < keep <= 1.:
+        fraction = 1. - keep
+    else:
+        raise ValueError("keep should be an integer >1 or a float [0-1]")
+    integrated_templates = _get_integrated_polar_templates(polar_image.shape[1], r_templates, intensities)
+    answer = _mixed_matching_lib_to_polar(polar_image, integrated_templates, r_templates,
+                                 theta_templates, intensities, fraction, n_best)
+    indices = answer[:, 0].astype(np.int64)
+    cors = answer[:, 1]
+    angles = answer[:, 2]
+    return indices, cors, angles
+
+
+@njit(["float64[:,:,:,:](float64[:,:,:,:], float64[:,:], int64[:,:], int64[:,:], float64[:,:], float64, int64)"],
+      nogil=True,
+      parallel=True,
+      )
+def _index_chunk(polar_images, integrated_templates, r_templates, theta_templates, intensities_templates, fraction, n_best):
+    indexation_result_chunk = np.empty((polar_images.shape[0], polar_images.shape[1], n_best, 3), dtype=np.float64)
+    for idx in prange(polar_images.shape[0]):
+        for idy in prange(polar_images.shape[1]):
+            pattern = polar_images[idx, idy]
+            # compute indexation_result
+            indexresult = _mixed_matching_lib_to_polar(pattern,
+                                                       integrated_templates,
+                                                       r_templates,
+                                                       theta_templates,
+                                                       intensities_templates,
+                                                       fraction,
+                                                       n_best
+                                                       )
+            indexation_result_chunk[idx, idy] = indexresult
+    return indexation_result_chunk
+
+
+def index_dataset_with_template_rotation(signal,
+                                         library,
+                                         phases,
+                                         optimize_direct_beam=True,
+                                         chunks="auto",
+                                         delta_r=1,
+                                         delta_theta=1,
+                                         n_best=1,
+                                         keep=100,
+                                         **kwargs):
+    workers = os.cpu_count()
+    result = {}
```

Comment:
The [`CrystalMap` class](https://orix.readthedocs.io/en/latest/reference.html#orix.crystal_map.crystal_map.CrystalMap) could be used instead of this dictionary to store values.

Reply:
The approach we've taken elsewhere is to do this conversion as late as possible. Generally with an `export_to_crystal_map` method of the `IndexationResult`

Reply:
I remember you saying something about this previously, but cannot remember why...

In kikuchipy, we even attach a CrystalMap with each dictionary of simulated patterns going into the pattern matching routines, storing the rotations, see e.g. [this part](https://github.com/pyxem/kikuchipy/blob/master/kikuchipy/indexing/_static_pattern_matching.py#L164).